indico/web/client/js/react/components/WTFDateTimeField.jsx

```diff
@@ -0,0 +1,234 @@
+  earliest,
+  latest,
+  defaultTime,
+  linkedField,
+}) {
+  const dateField = useMemo(() => document.getElementById(dateId), [dateId]);
+  const timeField = useMemo(() => document.getElementById(timeId), [timeId]);
+  const timezoneField = useMemo(() => document.getElementById(timezoneFieldId), [timezoneFieldId]);
+  const linkedFieldDateElem = useMemo(
+    () => linkedField && document.getElementById(`${linkedField.id}-datestorage`),
+    [linkedField]
+  );
+  const [time, setTime] = useState(timeField.value ? moment(timeField.value, 'HH:mm') : null);
+  const [date, setDate] = useState(dateField.value ? moment(dateField.value, 'DD/MM/YYYY') : null);
+  const [currentTimezone, setTimezone] = useState(timezone);
+  const earliestMoment = earliest ? moment(earliest) : null;
+  const latestMoment = latest ? moment(latest) : null;
+  const format = uses24HourFormat ? 'HH:mm' : 'hh:mm a';
+  const timePickerRef = useRef(null);
+  const clearRef = useRef();
+
+  const updateTime = useCallback(
+    value => {
+      timeField.value = value ? value.format('HH:mm') : '';
+      setTime(value);
+      trigger(timeId);
+    },
+    [timeField, timeId]
+  );
+
+  const updateDate = useCallback(
+    value => {
+      if (value && timeField.value === '') {
+        updateTime(moment(defaultTime, 'HH:mm'));
+      }
+      dateField.value = value ? value.format('DD/MM/YYYY') : '';
+      setDate(value);
+      trigger(dateId);
+      timePickerRef.current.picker.required = value;
+    },
+    [dateField, timeField, dateId, defaultTime, updateTime]
+  );
+
+  useEffect(() => {
+    if (!timezoneField) {
+      return;
+    }
+    function handleTimezone() {
+      setTimezone(timezoneField.value);
+    }
+    timezoneField.addEventListener('change', handleTimezone);
+
+    return () => {
+      timezoneField.removeEventListener('change', handleTimezone);
+    };
+  }, [timezoneField]);
+
+  useEffect(() => {
+    timePickerRef.current.picker.required = required;
+  }, [required]);
+
+  useEffect(() => {
+    if (!linkedField) {
+      return;
+    }
+    function handleDateChange() {
+      const linkedDate = moment(linkedFieldDateElem.value, 'DD/MM/YYYY');
+      if (
+        (linkedField.notBefore && linkedDate.isAfter(date, 'day')) ||
+        (linkedField.notAfter && linkedDate.isBefore(date, 'day'))
+      ) {
+        updateDate(linkedDate);
+      }
+    }
+    linkedFieldDateElem.addEventListener('change', handleDateChange);
+
+    return () => {
+      linkedFieldDateElem.removeEventListener('change', handleDateChange);
+    };
+  }, [date, linkedFieldDateElem, linkedField, updateDate]);
+
+  const clearFields = () => {
+    updateTime(null);
+    updateDate(null);
+    clearRef.current.dispatchEvent(new Event('indico:closeAutoTooltip'));
+  };
+
+  const getDisabledHours = useCallback(() => {
+    const hours = [];
+    if (earliestMoment && date.isSame(earliestMoment, 'day')) {
+      hours.push([...Array(earliestMoment.hour()).keys()]);
```

Comment:
`hours.push(_.range(earliestMoment.hour()))` would work as well (similar for the other cases below).
You'd just need to `import _ from 'lodash';`).

Feel free to leave it as-is though!